client/embedded-body-layout/embedded-body-layout.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { Router, Route, Switch } from 'react-router-dom';
+import { Suspense } from '@wordpress/element';
+import { useUser } from '@woocommerce/data';
+import { createBrowserHistory } from 'history';
+import { parse } from 'qs';
+
+/**
+ * Internal dependencies
+ */
+import { EmbeddedPage, embeddedPageRegistry } from './page-registry';
+import '../payments';
+import './style.scss';
+
+interface PrimaryLayoutProps {
+	page: EmbeddedPage;
+}
+
+const PrimaryLayout = ( { page }: PrimaryLayoutProps ) => {
+	return (
+		<div
+			className="woocommerce-embedded-layout__primary"
+			id="woocommerce-embedded-layout__primary"
+		>
+			<Suspense fallback={ null }>
+				<page.container />
+			</Suspense>
+		</div>
+	);
+};
+
+const customHistory = createBrowserHistory();
+Object.defineProperty( customHistory, 'location', {
+	get: () => {
+		const query = parse( location.search.substring( 1 ) ) as {
+			page: string;
+			tab: string;
+			section?: string;
+		};
+		let pathname = query.page + '_' + query.tab;
+		if ( query.section ) {
+			pathname += '_' + query.section;
+		}
+
+		return {
+			...location,
+			pathname,
+		};
+	},
+} );
```

Comment:
Overwriting the location param to create a path out of the query params `page`, `tab`, and `section`.